### ISSUE TITLE
Parameter renamed in custom resource template

### ISSUE THREAD

**Issue by diorman** [unbounce/iidy#105]
I'm having an issue trying to parameterize a redrive policy whithin a [custom resource template](https://github.com/unbounce/iidy#custom-resource-templates). iidy is prepending the name of the resource to the parameter (DeliveryDeadLetterQueue.Arn).

iidy version: 1.6.5

```
# delivery-queue-template.yaml

$params:
  ...
  - Name: RedrivePolicy
    Default: !Ref 'AWS::NoValue'
...

Resources:
  Queue:
    Type: AWS::SQS::Queue
    Properties:
      ...
      RedrivePolicy: !$ RedrivePolicy
```

```
# cfn-template.yaml

Resources:
  ...
  DeliveryDeadLetter:
    Type: DeliveryQueue
    Properties:
      EventType: DeliveryDeadLetter
  ...
  ProcessLeadEventV1:
    Type: DeliveryQueue
    Properties:
      ...
      RedrivePolicy:
        deadLetterTargetArn: !GetAtt DeliveryDeadLetterQueue.Arn
        maxReceiveCount: 2
```

Expected output:

```  
Resources:
  ...
  ProcessLeadEventV1Queue:
    Type: 'AWS::SQS::Queue'
    Properties:
      ...
      RedrivePolicy:
        deadLetterTargetArn: !GetAtt DeliveryDeadLetterQueue.Arn
        maxReceiveCount: 2
```

Current output:
```
Resources:
  ...
  ProcessLeadEventV1Queue:
    Type: 'AWS::SQS::Queue'
    Properties:
      ...
      RedrivePolicy:
        deadLetterTargetArn: !GetAtt ProcessLeadEventV1DeliveryDeadLetterQueue.Arn
        maxReceiveCount: 2
```

Is this the expected behavior? Is there a way to avoid renaming the parameter?

* Custom resource template: https://github.com/unbounce/lp-leads/blob/36f50f45c327509ed4cc0e64c207c055506be9b9/infrastructure/long-lived/delivery-queue-template.yaml
* cfn-template.yaml: https://github.com/unbounce/lp-leads/blob/36f50f45c327509ed4cc0e64c207c055506be9b9/infrastructure/long-lived/cfn-template.yaml

**Comment by tavisrudd** [comment]
Yes, the basic behaviour here is expected as you'd run into name clashes between multiple invocations of custom resource templates without it. However, we could make `iidy` a bit smarter to avoid prefixing values that have been passed in as custom template parameters. 

https://github.com/unbounce/iidy/blob/b60598e67f78f1d43ed1a25c5d33627d62b7b685/src/preprocess/index.ts#L1076 is part of the code that would need modification in addition to something near https://github.com/unbounce/iidy/blob/b60598e67f78f1d43ed1a25c5d33627d62b7b685/src/preprocess/index.ts#L740.

**Comment by tavisrudd** [comment]
Actually, the fix might be simpler than that. The `!GetAtt` call is getting visited twice. If we escape it after the first call to `visitNode` (on https://github.com/unbounce/iidy/blob/b60598e67f78f1d43ed1a25c5d33627d62b7b685/src/preprocess/index.ts#L740) the second call on https://github.com/unbounce/iidy/blob/b60598e67f78f1d43ed1a25c5d33627d62b7b685/src/preprocess/index.ts#L759 would leave it alone.

**Comment by tavisrudd** [comment]
This is fixed by #233.
